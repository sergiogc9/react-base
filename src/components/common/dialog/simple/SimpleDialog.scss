@import "@src/scss/_globals.scss";

.react-base-simple-dialog {
	> div {
		width: 40rem;
	}

	.md-dialog-content {
		padding: 0.5rem 1rem;
	}

	.dialog-container {
		.dialog-header {
			margin-bottom: 2rem;
			margin-top: 0.5rem;
			padding: 1rem;

			.dialog-text {
				line-height: 2.8rem;
				font-size: 2rem;
			}
		}

		.dialog-content {
			padding: 0 1rem;
			width: 100%;
			color: #757575;
			font-size: 1.4rem;
		}

		.dialog-footer {
			margin-top: 4rem;
			justify-content: space-between;
			padding: 0.8rem 0;

			&.center-buttons {
				justify-content: center;
			}

			button {
				border-radius: 3px;
				font-size: 1.2rem;

				&.btn--grey {
					background-color: $md-grey-300;
				}

				&.btn--green {
					background-color: $react-base-green;
				}
			}
		}
	}
}
