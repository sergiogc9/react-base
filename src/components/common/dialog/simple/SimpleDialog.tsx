import React from 'react';
import DialogContainer from 'react-md/lib/Dialogs/DialogContainer';
import Button from 'react-md/lib/Buttons/Button';

import { withT } from '@src/lib/i18n';
import { ComponentProps } from './types';

import './SimpleDialog.scss';

const SimpleDialog = (props: ComponentProps) => {

	const { id, text, content, onAccept, onAcceptText, onCancel, onCancelText } = props;

	const centerButtons = (!onAccept && onCancel) || (onAccept && !onCancel) ? "center-buttons" : "";
	const footerClassName = `dialog-footer md-grid ${centerButtons}`;

	const dialogContent = (
		<div className="dialog-container">
			<div className="dialog-header md-grid">
				<div className="dialog-text">{text}</div>
			</div>
			<div className="dialog-content">{content}</div>
			<div className={footerClassName}>
				{onCancel ? <Button id="react-baseDialogCancelBtn" className="btn--grey btn--no-background" flat onClick={onCancel}>{onCancelText}</Button> : null}
				{onAccept ? <Button id="react-baseDialogAcceptBtn" className="btn--green btn--no-background" flat onClick={onAccept}>{onAcceptText}</Button> : null}
			</div>
		</div>
	);

	return <DialogContainer
		id={id}
		className="react-base-simple-dialog"
		visible={true}
		onHide={onCancel ? onCancel : () => { }}
		focusOnMount={false}
		aria-labelledby="react-base-dialog"
		children={dialogContent}
	/>;
};

export default withT(SimpleDialog);
