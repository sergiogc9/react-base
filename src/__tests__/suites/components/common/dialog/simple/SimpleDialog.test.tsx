import { shallow } from 'enzyme';
import React from 'react';

import SimpleDialog from '@src/components/common/dialog/simple';

let onAcceptMock: jest.Mock;
let onCancelMock: jest.Mock;

describe('<Dialog />', () => {
	let wrapper: any;

	beforeEach(() => {
		onAcceptMock = jest.fn();
		onCancelMock = jest.fn();
		wrapper = shallow((<SimpleDialog
			id="dialog-id"
			text="Dialog title"
			content={(<div id="dialog-content"></div>)}
			onAccept={onAcceptMock}
			onAcceptText="Accept"
			onCancel={onCancelMock}
			onCancelText="Cancel"
		></SimpleDialog>));
	});

	it("Test Dialog component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find('#dialog-content')).toHaveLength(1);
		expect(wrapper.find('#react-baseDialogAcceptBtn')).toHaveLength(1);
		expect(wrapper.find('#react-baseDialogCancelBtn')).toHaveLength(1);
	});

	it("Test Dialog without accept button", () => {
		wrapper.setProps({ onAccept: undefined });
		expect(wrapper.find('#react-baseDialogAcceptBtn')).toHaveLength(0);
	});

	it("Test Dialog without cancel button", () => {
		wrapper.setProps({ onCancel: undefined });
		expect(wrapper.find('#react-baseDialogCancelBtn')).toHaveLength(0);
	});

	it("Test Dialog footer centered buttons class", () => {
		wrapper.setProps({ onCancel: undefined }); // onAccept is set
		expect(wrapper.find('.dialog-footer').hasClass('center-buttons')).toBe(true);
	});
});
