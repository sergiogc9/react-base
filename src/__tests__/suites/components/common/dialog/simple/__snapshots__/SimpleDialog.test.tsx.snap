// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> Test Dialog component snapshot 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover react-base-simple-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"react-base-dialog\\" id=\\"dialog-id\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-text\\">Dialog title</div></div><div class=\\"dialog-content\\"><div id=\\"dialog-content\\"></div></div><div class=\\"dialog-footer md-grid \\"><button id=\\"react-baseDialogCancelBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--grey btn--no-background\\"><div class=\\"md-ink-container\\"></div>Cancel</button><button id=\\"react-baseDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>Accept</button></div></div></section></div></span>"`;
